### PR TITLE
fix: perform basic form validation before address verification

### PR DESCRIPTION
**Pull request by starsirius**
The type of this PR is: **Fix**

This PR solves [EMI-1316](https://artsyproduct.atlassian.net/browse/EMI-1316)

### Description

Basic form validation (e.g. required fields) should be performed before the address verification (e.g. calling Smarty API for suggestions). This changes the order to perform basic form validation first.

### Before and after
<details><summary>Before</summary>
<p>

form-validation-first

</p>
</details> 

<details><summary>After</summary>
<p>

form-validation-first-after

</p>
</details> 

[EMI-1316]: https://artsyproduct.atlassian.net/browse/EMI-1316?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/Apps/Order/Routes/__tests__/Shipping.jest.tsx [hunk 1]
```diff
@@ -1029,6 +1029,38 @@ describe("Shipping", () => {
...
+            await page.clickSubmit()
+
+            const input = page
+              .find(Input)
+              .filterWhere(wrapper => wrapper.props().title === "Full name")
+            expect(input.props().error).toEqual("This field is required")
+
+            expect(
+              page.find(`[data-testid="address-verification-flow"]`).exists()
+            ).toBe(false)
```
Comment (starsirius) [inline src/Apps/Order/Routes/__tests__/Shipping.jest.tsx:1061]: This test is able to [catch the issue](https://app.circleci.com/pipelines/github/artsy/force/50599/workflows/b13476cc-ea46-40d4-beae-edf02f111ad1/jobs/379197)! 🎉 

rerun-causality-sync

File: src/Apps/Order/Routes/__tests__/Shipping.jest.tsx [hunk 2]
```diff
@@ -1626,9 +1658,10 @@ describe("Shipping", () => {
             )
             const wrapper = getWrapper({
               CommerceOrder: () => UntouchedBuyOrderWithShippingQuotes,
-              Me: () => emptyTestMe,
+              Me: () => testMe,
```
Comment (starsirius) [inline src/Apps/Order/Routes/__tests__/Shipping.jest.tsx:1661]: I think the previous setup wasn't right so the test started to fail after my change. This fixes it.

File: src/Apps/Order/Routes/Shipping/index.tsx [hunk 3]
```diff
@@ -257,28 +292,7 @@ export const ShippingRoute: FC<ShippingProps> = props => {
   }
 
   const selectShipping = async (editedAddress?: MutationAddressResponse) => {
-    if (shippingOption === "SHIP") {
-      if (isCreateNewAddress()) {
-        // validate when order is not pickup and the address is new
-        const { errors, hasErrors } = validateAddress(address)
-        const { error, hasError } = validatePhoneNumber(phoneNumber)
```
Comment (starsirius) [inline src/Apps/Order/Routes/Shipping/index.tsx:264]: Previously the basic form validation was in `finalizeFulfillment()` which is after address verification. I moved it out to before the address verification.

**Review by leamotta — APPROVED** [review]

**Comment by starsirius** [comment]
Merging this in preparation of the QA! 💪

**Pull request merged**